### PR TITLE
gate.io: Enforce the use of 'convert.StringToFloat64' and permit its use in outbound requests

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Randomly errors: `json: invalid use of ,string struct tag, trying to unmarshal "" into float64`

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by thrasher- — APPROVED** [review]
LGTM, nice!

File: common/convert/convert.go [hunk 1]
```diff
@@ -222,6 +222,15 @@ func (f *StringToFloat64) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// MarshalJSON implements the json.Marshaler interface.
+func (f StringToFloat64) MarshalJSON() ([]byte, error) {
+	if f == 0 {
+		return []byte(jsonStringIdent + jsonStringIdent), nil
+	}
```
Comment (gloriousCode) [inline common/convert/convert.go:229]: Hope no one ever wants to send "0" (but they have other options anyway)

**Review by gloriousCode — APPROVED** [review]
tACK. No gateio usecases come with a shock if there is a marshalled 0 StringToFloat64 in the request

File: common/convert/convert.go (hunk 1 above)
Comment (shazbert) [reply common/convert/convert.go:229]: Panic() is an option.

**Pull request merged**